state/application.go

```diff
@@ -1005,6 +1005,60 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	return nil
 }
 
+// UpdateApplicationSeries updates the series for the Application.
+func (a *Application) UpdateApplicationSeries(series string) (err error) {
+	// Make sure no units are added or removed while the update-series
+	// transaction is being executed. To cover the case where a new subordinate
+	// may be added.
+	units, err := a.AllUnits()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	var unitOps []txn.Op
```

Comment:
These ops all need to be built up inside a buildTxn() function, with matching business logic to perform the same checks as what the asserts do.
(as an aside, there's no model is alive check and there needs to be).